contents/tutorials/nuxt-analytics.md

```diff
@@ -0,0 +1,285 @@
+
+```js file=plugins/posthog.client.js
+import { defineNuxtPlugin } from '#app'
+import posthog from 'posthog-js'
+
+export default defineNuxtPlugin(nuxtApp => {
+  const runtimeConfig = useRuntimeConfig();
+  const posthogClient = posthog.init(runtimeConfig.public.posthogPublicKey, {
+    api_host: runtimeConfig.public.posthogHost,
+  })
+  
+  return {
+    provide: {
+      posthog: () => posthogClient
+    }
+  }
+})
+```
+
+Once you’ve done this, reload your app. You should begin seeing events in the [PostHog events explorer](https://us.posthog.com/events).
+
+<ProductScreenshot
+  imageLight={EventsLight} 
+  imageDark={EventsDark} 
+  alt="Events in PostHog" 
+  classes="rounded"
+/>
+
+### Capturing pageviews
+
+You might notice that moving between pages only captures a single pageview event. This is because PostHog only captures pageview events when [page load](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event) is fired. Since Nuxt creates a single-page app, this only happens once, and the Nuxt router handles subsequent page changes.
+
+If we want to capture every route change, we must write code to capture pageviews that integrates with the router.
+
+In `posthog.client.js`, set up PostHog to capture pageviews in the `router.afterEach` function. Additionally, you can use `nextTick` so that the capture event fires only after the page is mounted.
+
+```js plugins/posthog.client.js
+import { defineNuxtPlugin } from '#app'
+import posthog from 'posthog-js'
+
+export default defineNuxtPlugin(nuxtApp => {
+  const runtimeConfig = useRuntimeConfig();
+  const posthogClient = posthog.init(runtimeConfig.public.posthogPublicKey, {
+    api_host: runtimeConfig.public.posthogHost,
+    capture_pageview: false // set this to false since we manually capture pageviews in router.afterEach
+  })
+
+  const router = useRouter();
+  router.afterEach((to) => {
+    nextTick(() => {
+      posthog.capture('$pageview', {
+        current_url: to.fullPath
+      });
+    });
+  });
+
+  // rest of your code
+})
+
+```
+
+Make sure to set `capture_pageview` in the PostHog initialization config to `false`. This turns off autocaptured pageviews and ensures you won’t double-capture pageviews on the first load.
+
+### Capturing custom events
+
+Beyond pageviews, there might be more events you want to capture. You can do this by capturing custom events with PostHog. 
+
+To showcase this, we add a button to the home page and capture a custom event whenever it is clicked. Update the code in `index.vue`:
+
+```vue file=HomePage.vue
+<template>
+  <div>
+    <h1>Home</h1>
+    <NuxtLink to="/about">Go to About</NuxtLink>
+    <button @click="captureCustomEvent">Click Me</button>
+  </div>
+</template>
+
+<script setup>
+import { usePostHog } from 'path-to-your-posthog-plugin';
+
+const postHog = usePostHog();
+
+const captureCustomEvent = () => {
+  postHog.capture('home_button_clicked', {
+    'user_name': 'Max the Hedgehog' 
+  });
+};
+</script>
```

Comment:
This component isn't working, even when pointing `import { usePostHog } from 'path-to-your-posthog-plugin';` to the correct file.

Reply:
fixed